node/src/reactor/main_reactor/keep_up.rs

```diff
@@ -393,7 +544,20 @@ impl MainReactor {
                 debug!(%era_id, "historical: era already present or is not relevant");
             }
         }
-        KeepUpInstruction::CheckLater("historical sync leap received".to_string(), Duration::ZERO)
+
+        // We put all the headers we received into storage - if the SyncLeap was valid, we can
+        // trust them, and they will be useful for determining which blocks are immediate switch
+        // blocks.
+        let effects = best_available
+            .headers()
+            .flat_map(move |block_header| {
+                effect_builder
+                    .put_block_header_to_storage(Box::new(block_header.clone()))
```

Comment:
Isn't it already stored by the sync leap fetcher [here](https://github.com/casper-network/casper-node/blob/dev/node/src/components/fetcher/fetcher_impls/sync_leap_fetcher.rs#L41-L60)?

Reply:
Ha, good point. It was kind of weird to me that they wouldn't be stored, so it makes sense that they actually would :smile: